packages/mds-db/devices.ts

```diff
@@ -130,9 +114,16 @@ export async function updateDevice(device_id: UUID, provider_id: UUID, changes:
   const res = await client.query(sql, values)
 
   if (res.rowCount === 0) {
-    throw new Error('not found')
+    throw new NotFoundError('not found')
   } else {
-    return readDevice(device_id, provider_id)
+    const device = IngestServiceClient.getDevice({ device_id, provider_id })
+    if (!isDefined(device)) {
+      throw new NotFoundError(`device_id ${device_id} not found`)
+    }
+
+    // Cast is necessary because the compiler is not smart enough to understand that
+    // the if statement above checks whether `device` is `undefined`
+    return device as unknown as DeviceDomainModel
```

Comment:
So this `as` cast is not actually necessary. This casting is actually covering up a bug, where on line 119 we're missing an `await`! The type for `device` here is a `Promise<DeviceDomainModel | Undefined>`, so we need to do an await first. That's why the compiler was barfing, not because the isDefined check wasn't doing what it should.
```suggestion
    const device = await IngestServiceClient.getDevice({ device_id, provider_id })
    if (!isDefined(device)) {
      throw new NotFoundError(`device_id ${device_id} not found`)
    }

    return device
```